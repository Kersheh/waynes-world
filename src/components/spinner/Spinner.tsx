import React from 'react';
import classNames from 'classnames/bind';

import styles from './Spinner.module.scss';

const cx = classNames.bind(styles);

interface SpinnerProps {
  isFullscreen?: boolean;
  show?: boolean;
}
const Spinner = ({ isFullscreen = true, show = true }: SpinnerProps) => {
  return (
    <>
      {show && (
        <div
          className={cx({
            spinner: true,
            isFullscreen
          })}
        >
          <div className={styles.spinnerIcon} />
        </div>
      )}
    </>
  );
};

export default Spinner;
