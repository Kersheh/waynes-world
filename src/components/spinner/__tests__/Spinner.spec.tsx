import React from 'react';
import { render } from '@testing-library/react';

import Spinner from '../Spinner';

describe('<Spinner/> component', () => {
  it('should render', () => {
    const { container } = render(<Spinner />);

    expect(container).toMatchSnapshot();
  });
});
