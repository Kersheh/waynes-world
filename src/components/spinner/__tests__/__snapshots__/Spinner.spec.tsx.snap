// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner/> component should render 1`] = `
<div>
  <div
    class="spinner isFullscreen"
  >
    <div
      class="spinnerIcon"
    />
  </div>
</div>
`;
